graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/Federation.java

```diff
@@ -154,14 +163,16 @@ private static RuntimeWiring ensureFederationDirectiveDefinitionsExist(
         typeRegistry.add(link__Import.definition);
       }
       if (!runtimeWiring.getScalars().containsKey(link__Import.typeName)) {
-        newRuntimeWiring = copyRuntimeWiring(newRuntimeWiring).scalar(link__Import.type).build();
+        newRuntimeWiring =
+            copyRuntimeWiring(newRuntimeWiring, Collections.singleton(link__Import.type));
       }
     }
 
     return newRuntimeWiring;
   }
 
-  private static RuntimeWiring.Builder copyRuntimeWiring(RuntimeWiring runtimeWiring) {
+  private static RuntimeWiring copyRuntimeWiring(
+      RuntimeWiring runtimeWiring, Set<GraphQLScalarType> additionalScalars) {
```

Comment:
I agree about the method returning a Builder - it fits better with the rest of the code, but unfortunately I don't see a way to add the typeResolver on the `RuntimeWiring.Builder` the same way you can do it on `RuntimeWiring` directly (it looks like `RuntimeWiring.Builder.type(TypeRuntimeWiring typeRuntimeWiring)` has some side effects on `defaultDataFetchers`, I'm unsure what the impacts of touching it could be).

Agreed on a `RuntimeWiring.newRuntimeWiring(existingRuntimeWiring)`! I can open a bug and/or a PR.

Reply:
Gotcha, I missed how this was required for the `typeResolver` thing 👍